### PR TITLE
Add io.github.dxx_rebirth.dxx_rebirth.d1x-rebirth

### PR DESCRIPTION
**Pull request by fpiesche**
### Please confirm your submission meets all the criteria

- [x] Please describe the application briefly.
  DXX-Rebirth is a modern source port of Descent/Descent II. D1X-Rebirth is the Descent 1 variant of this.

- [x] The domain used for the application ID is [controlled by the application developers][appid-domain] and the [application id guidelines][appid] are followed.
- [x] I have read the and followed all the [Submission and licence requirements][reqs].
- [x] I have [built][build] and tested the submission locally.
- [x] I am an author/developer/upstream contributor of the project. If not, I contacted upstream developers about this submission. **Link:** https://github.com/dxx-rebirth/dxx-rebirth/issues/691#issuecomment-2237823331

[appid-domain]: https://docs.flathub.org/docs/for-app-authors/requirements#control-over-domain
[appid]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id
[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission

### CODE REVIEW

File: net.sourceforge.scons.scons.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+name: net.sourceforge.scons.scons
```
Comment (hfiguiere) [inline net.sourceforge.scons.scons.yml:1]: there is absolutely no reason to have this in a sperate manifest file.

File: net.sourceforge.scons.scons.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    url: https://downloads.sourceforge.net/project/scons/scons/4.6.0/SCons-4.6.0.tar.gz
+    sha256: 7db28958b188b800f803c287d0680cc3ac7c422ed0b1cf9895042c52567803ec
+    x-checker-data:
+      type: anitya
+      project-id: 4770
+      stable-only: true
+      url-template: https://downloads.sourceforge.net/project/scons/scons/$version/SCons-$version.tar.gz
+build-commands:
+  - pip3 install --no-index --no-build-isolation --prefix=/app .
+cleanup:
```
Comment (hfiguiere) [inline net.sourceforge.scons.scons.yml:14]: the cleanup rule should be `'*'`

File: net.sourceforge.scons.scons.yml [hunk 3]
```diff
@@ -0,0 +1,18 @@
+name: net.sourceforge.scons.scons
+buildsystem: simple
+sources:
+  - type: archive
+    url: https://downloads.sourceforge.net/project/scons/scons/4.6.0/SCons-4.6.0.tar.gz
+    sha256: 7db28958b188b800f803c287d0680cc3ac7c422ed0b1cf9895042c52567803ec
+    x-checker-data:
```
Comment (hfiguiere) [inline net.sourceforge.scons.scons.yml:7]: why do you need to autoupdate it? It's a build tool

File: io.github.dxx_rebirth.d1x-rebirth.yml [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+finish-args:
+  # hardware 3D and controller access
+  - --device=all
+  # X11 + XShm access
+  - --share=ipc
+  - --socket=x11
+  # Audio
+  - --socket=pulseaudio
+  # Give permission to use existing setup if available
+  - --filesystem=~/.d1x-rebirth
```
Comment (hfiguiere) [inline io.github.dxx_rebirth.d1x-rebirth.yml:16]: no. it should jsut persist.

File: io.github.dxx_rebirth.d1x-rebirth.yml [hunk 5]
```diff
@@ -0,0 +1,63 @@
+app-id: io.github.dxx_rebirth.d1x-rebirth
```
Comment (bbhtt) [inline io.github.dxx_rebirth.d1x-rebirth.yml:1]: This ID won't work. There is no repository at https://github.com/dxx-rebirth/d1x-rebirth

Can you change it to something like `io.github.dxx_rebirth.dxx_rebirth.d2x-rebirth` ?

See the control over domain portion here https://docs.flathub.org/docs/for-app-authors/requirements/#control-over-domain

Comment (bbhtt) [reply io.github.dxx_rebirth.d1x-rebirth.yml:1]: Repeating is fine, see the second example here https://docs.flatpak.org/en/latest/conventions.html#application-ids

File: io.github.dxx_rebirth.d1x-rebirth.yml [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+  # Audio
+  - --socket=pulseaudio
+  # Give permission to use existing setup if available
+  - --filesystem=~/.d1x-rebirth
+  # Enable network for multiplayer
+  - --share=network
+
+modules:
+  - shared-modules/glu/glu-9.json
+  - shared-modules/physfs/physfs.json
```
Comment (bbhtt) [inline io.github.dxx_rebirth.d1x-rebirth.yml:22]: You forgot to add shared-modules.

**Comment by bbhtt** [comment]
bot, build io.github.dxx_rebirth.dxx_rebirth.d1x-rebirth

**Pull request closed without merging**